### ISSUE TITLE
[18.0] sale: Incorrect tax rounding

### ISSUE THREAD

**Issue by Levi-Bensley** [odoo/odoo#182161]
### Odoo Version

- [ ] 15.0
- [ ] 16.0
- [ ] 17.0
- [X] Other (specify)

### Steps to Reproduce

* Create a combo product with the sales price of 1.00, tax 20%
* Add combos that are the same prices.
* Add combo product to sale order via sales.

**Current Behaviour:**
Gives inconsistent tax rounding depending on the amount of combo items.
1 combo item = 1.20 (Inc. Tax) // Correct
2 combo items = 1.20 // Correct
3 combo items = 1.21 // **Incorrect**
4 combo items = 1.20 // Correct
5 combo items = 1.20 // Correct
6 combo items = 1.18 // **Incorrect**
7 combo items = 1.21 // **Incorrect**
8 combo items = 1.23 // **Incorrect**
9 combo items = 1.18 // **Incorrect**

**Expected Behaviour:**
All of them should be 1.20 including tax.
Maybe would be better to have the base price on the main product row then the combo items either have 0 or price extra?

**Comment by ltinel** [comment]
Hi, combos started to be supported in the sales app in version 18. Could you add more details about the issues you're getting in version 16 and 17? Maybe the issue is more general and not related to combos.

**Comment by Levi-Bensley** [comment]
I forgot to update the title, it seems to just happen generically to the price split of, 0.33, 0.33, 0.34.

So I can see how the rounding error is happening with these numbers. But yeah it's is generic issue with rounding the lines.

**Comment by Levi-Bensley** [comment]
However, further looking, with pre v18, this may be fine in the UK for tax per item, not 100% sure though. However, for a combo item, it feels like the tax should probably be based on the price set on the combo item, not split across it's items?

**Comment by ltinel** [comment]
For combos, the tax is indeed applied on the combo item lines, not on the "main" combo line (which has a price of 0 anyway).

On another topic, I couldn't reproduce your issue in version 18. Would you mind sharing a screencast?

**Comment by Levi-Bensley** [comment]
This is on the current v18 runbot. [Screenshot](https://ibb.co/XFM4P6J)

**Comment by ltinel** [comment]
Thanks for the screenshot, I'll forward the issue to the appropriate team.

**Comment by Levi-Bensley** [comment]
Just further information to this, if you have two identical combos with this pricing structure, but on separate lines, you will also get a total of 2.42. Not an issue if the current round is correct. However, if you have a combo with this pricing structure, but set the quantity to 2 instead of adding it as a new line, it will be 2.40.

**Comment by Levi-Bensley** [comment]
How I (and a few people i've asked) think the combos and their pricing should be handled is like 
![Screenshot-2024-10-21-133138](https://github.com/user-attachments/assets/7d353bfe-7a4c-41a3-8fb6-01c5ddf9e36f)

**Comment by ltinel** [comment]
You mean that you'd like the tax to be applied on the combo product (main line) and not the combo items (sublines)? We decided not to do this as combo items can have different taxes. Going back to your example, if you chose a can of beer instead of a coke, the tax would probably be different.

**Comment by Levi-Bensley** [comment]
True, I suppose with the beer. However, I guess the bigger issue currently is with the inconsistency mentioned in my message before that.
